release-calendar/src/client/components/Calendar.tsx

```diff
@@ -25,11 +28,36 @@ const CALENDAR_CONTENT_HEIGHT = 480;
 const EVENT_LIMIT_DISPLAYED = 3;
 
 export interface CalendarProps {
-  events: EventSourceInput[];
+  channels: Channel[];
 }
 
-export class Calendar extends React.Component<CalendarProps, {}> {
+export interface CalendarState {
+  events: Map<Channel, EventSourceInput>;
+}
+
+export class Calendar extends React.Component<CalendarProps, CalendarState> {
+  private apiService: ApiService;
+
+  constructor(props: Readonly<CalendarProps>) {
+    super(props);
+    this.state = {
+      events: new Map<Channel, EventSourceInput>(),
+    };
+    this.apiService = new ApiService();
+  }
+
+  async componentDidMount(): Promise<void> {
+    const releases = await this.apiService.getReleases();
+    this.setState({events: getEvents(releases)});
+  }
+
   render(): JSX.Element {
+    const displayedEvents: EventSourceInput[] = [];
+    this.props.channels.forEach((channel) => {
+      if (this.state.events.has(channel)) {
+        displayedEvents.push(this.state.events.get(channel));
```

Comment:
Another way of doing this:
```ts
    const displayEvents: EventSourceInput[] = this.props.channels
        .filter((channel) => this.state.events.has(channel))
        .map((channel) => this.state.events.get(channel));
```

(your way is fine too! I'm just personally a fan of functional programming)

Reply:
Oh, wait, yes, I was wondering how to make this better! filter and then map, smart.